File: backend/internal/server/routes/packages.go
package routes

import (
	"github.com/Wei-Shaw/sub2api/internal/handler"
	"github.com/Wei-Shaw/sub2api/internal/server/middleware"
	"github.com/Wei-Shaw/sub2api/internal/service"
	"github.com/gin-gonic/gin"
)

func RegisterPackageRoutes(v1 *gin.RouterGroup, h *handler.PackageHandler, jwt middleware.JWTAuthMiddleware, admin middleware.AdminAuthMiddleware, audit middleware.AuditLogMiddleware, settings *service.SettingService, limits *middleware.PanelRateLimiter) {
	user := v1.Group("/packages", gin.HandlerFunc(jwt), middleware.BackendModeUserGuard(settings), limits.Global())
	user.GET("/plans", h.ListPlans)
	user.GET("/mine", h.Mine)
	user.PUT("/order", h.Reorder)
	user.GET("/groups", h.ListGroups)
	user.POST("/groups", h.CreateGroup)
	user.GET("/groups/:id", h.Group)
	admins := v1.Group("/admin/packages", gin.HandlerFunc(admin), gin.HandlerFunc(audit), middleware.AdminComplianceGuard(settings))
	admins.GET("/plans", h.AdminListPlans)
	admins.POST("/plans", h.SavePlan)
	admins.PUT("/plans/:id", h.SavePlan)
	admins.DELETE("/plans/:id", h.StopSale)
}

File: frontend/src/api/packages.ts
import { apiClient } from './client'
import type { PackageGroupBuy, PackagePlan, UserPackage } from '@/types/packages'

export const packagesAPI = {
  async plans() { return (await apiClient.get<PackagePlan[]>('/packages/plans')).data },
  async mine() { return (await apiClient.get<UserPackage[]>('/packages/mine')).data },
  async reorder(packageIDs: number[]) {
    await apiClient.put('/packages/order', { package_ids: packageIDs })
  },
  async groups() { return (await apiClient.get<PackageGroupBuy[]>('/packages/groups')).data },
  async group(id: number) { return (await apiClient.get<PackageGroupBuy>(`/packages/groups/${id}`)).data },
  async startGroup(planID: number) {
    return (await apiClient.post<PackageGroupBuy>('/packages/groups', { plan_id: planID })).data
  },
  async adminPlans() { return (await apiClient.get<PackagePlan[]>('/admin/packages/plans')).data },
  async savePlan(plan: PackagePlan) {
    return plan.id
      ? (await apiClient.put<PackagePlan>(`/admin/packages/plans/${plan.id}`, plan)).data
      : (await apiClient.post<PackagePlan>('/admin/packages/plans', plan)).data
  },
  async unpublish(id: number) { await apiClient.delete(`/admin/packages/plans/${id}`) }
}
